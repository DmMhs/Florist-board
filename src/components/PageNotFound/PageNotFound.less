.PageNotFound {
  min-height: calc(100vh - 7rem);
  border: 1px solid transparent;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  h2 {
    color: #333333;
    font-weight: normal;
    i {
      font-size: 2.5rem;
    }
  }
}
