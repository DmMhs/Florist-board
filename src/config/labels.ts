const labels = {
  brand: 'FLORIST.UA',
  adress: 'MARKET SQUARE, LVIV, UA',
  postCode: '346346',
  phone: '+123-45-678',
  navigation: {
    home: 'HOME',
    shop: 'SHOP',
    gallery: 'GALLERY',
    contacts: 'CONTACTS',
    cart: 'CART',
    account: {
      main: 'MY ACCOUNT',
      menu: {
        signIn: 'SIGN IN',
        signUp: 'SIGN UP'
      }
    }
  }
};

export default labels;
