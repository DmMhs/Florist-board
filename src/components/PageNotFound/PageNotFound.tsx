import React, { useContext } from 'react';

import './PageNotFound.less';
import { AppContext } from '../../AppContext';
import labels from '../../config/labels';

const pageNotFound = () => {
  const context = useContext(AppContext);
  return (
    <div className="PageNotFound">
      <h2>
        <span className="message">
          {context.state.lang === 'en'
            ? labels.en.pages.pageNotFound
            : labels.uk.pages.pageNotFound}
        </span>{' '}
        <i className="far fa-grimace" />
      </h2>
    </div>
  );
};

export default pageNotFound;
