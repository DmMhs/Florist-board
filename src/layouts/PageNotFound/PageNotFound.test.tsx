import React from 'react';
import { shallow, mount } from 'enzyme';

import PageNotFound from './PageNotFound';
import AppContextProvider from '../../AppContext';
import { BrowserRouter } from 'react-router-dom';
import labels from '../../config/labels';

describe('PageNotFound works as expected', () => {
  it('PageNotFound component matches a snapshot', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <PageNotFound />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('Message is not empty', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <PageNotFound />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();

    expect(wrapper.find('.message').text().length).toBeGreaterThan(0);
  });
});
