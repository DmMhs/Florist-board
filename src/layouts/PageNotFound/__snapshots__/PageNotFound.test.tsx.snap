// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound works as expected PageNotFound component matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <pageNotFound>
            <div
              className="PageNotFound"
            >
              <h2>
                <span
                  className="message"
                >
                  Page not found
                </span>
                 
                <i
                  className="far fa-grimace"
                />
              </h2>
            </div>
          </pageNotFound>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
